framework/encode/vulkan_capture_common.cpp

```diff
@@ -0,0 +1,260 @@
+/*
+ ** Copyright (c) 2024 LunarG, Inc.
+ **
+ ** Permission is hereby granted, free of charge, to any person obtaining a
+ ** copy of this software and associated documentation files (the "Software"),
+ ** to deal in the Software without restriction, including without limitation
+ ** the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ ** and/or sell copies of the Software, and to permit persons to whom the
+ ** Software is furnished to do so, subject to the following conditions:
+ **
+ ** The above copyright notice and this permission notice shall be included in
+ ** all copies or substantial portions of the Software.
+ **
+ ** THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ ** IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ ** FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ ** AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ ** LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ ** FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ ** DEALINGS IN THE SOFTWARE.
+ */
+
+#include "vulkan_capture_common.h"
+
+#if defined(VK_USE_PLATFORM_ANDROID_KHR)
+#include <android/hardware_buffer.h>
+#endif
+
+GFXRECON_BEGIN_NAMESPACE(gfxrecon)
+GFXRECON_BEGIN_NAMESPACE(encode)
+
+void CommonWriteCreateHardwareBufferCmd(format::ThreadId                                    thread_id,
+                                        format::HandleId                                    memory_id,
+                                        AHardwareBuffer*                                    hardware_buffer,
+                                        const std::vector<format::HardwareBufferPlaneInfo>& plane_info,
+                                        VulkanCaptureManager*                               vulkan_capture_manager,
+                                        VulkanStateWriter*                                  vulkan_state_writer)
+{
+#if defined(VK_USE_PLATFORM_ANDROID_KHR)
```

Comment:
Just wondering: as this file is mostly android-only code, can it be conditionally included in the source list at cmake configuration-time (when configuring for Android)?

Reply:
It could be, but I think in the future more function should be added here, including code that is not android specific. If you look at the `vulkan_state_writer.cpp` there are many TODOs saying it is common code and should be combined